src/views/charts/TradingRewardsChart.tsx

```diff
@@ -0,0 +1,238 @@
+    shallowEqual
+  );
+
+  useEffect(() => {
+    // Initialize daily data for rewards chart
+    abacusStateManager.setHistoricalTradingRewardPeriod(HistoricalTradingRewardsPeriod.DAILY);
+  }, [canViewAccount]);
+
+  const msForPeriod = useCallback(
+    (period: TradingRewardsPeriod) => {
+      switch (period) {
+        case TradingRewardsPeriod.Period1d:
+          return 1 * timeUnits.day;
+        case TradingRewardsPeriod.Period7d:
+          return 7 * timeUnits.day;
+        case TradingRewardsPeriod.Period30d:
+          return 30 * timeUnits.day;
+        case TradingRewardsPeriod.Period90d:
+          return 90 * timeUnits.day;
+        case TradingRewardsPeriod.PeriodAllTime:
+        default:
+          return (now - Number(rewardsHistoryStartDate) + 1) * timeUnits.day;
+      }
+    },
+    [now, rewardsHistoryStartDate]
+  );
+
+  // Unselect selected period in toggle if user zooms in/out
+  const onZoomSnap = useMemo(
+    () =>
+      debounce(({ zoomDomain }: { zoomDomain?: number }) => {
+        if (zoomDomain) {
+          setIsZooming(!tradingRewardsPeriods.map(msForPeriod).includes(zoomDomain));
+        }
+      }, 200),
+    [msForPeriod]
+  );
+
+  const onToggleInteract = () => setIsZooming(false);
+
+  const rewardsData = useMemo(
+    () =>
+      periodTradingRewards && canViewAccount
+        ? periodTradingRewards
+            .toArray()
+            .reverse()
+            .map(
+              (datum) =>
+                ({
+                  date: new Date(datum.endedAtInMilliseconds).valueOf(),
+                  cumulativeAmount: datum.cumulativeAmount,
+                } as TradingRewardsDatum)
+            )
+        : [],
+    [periodTradingRewards, canViewAccount]
+  );
+
+  return (
+    <>
+      {rewardsData.length === 0 ? undefined : (
+        <$TitleContainer>
+          <$Title>
+            {stringGetter({
+              key: STRING_KEYS.TRADING_REWARDS,
+            })}
+          </$Title>
+
+          <ToggleGroup
+            items={tradingRewardsPeriods.map((period) => ({
+              value: period,
+              label:
+                period === TradingRewardsPeriod.PeriodAllTime
+                  ? stringGetter({ key: STRING_KEYS.ALL })
+                  : formatRelativeTime(msForPeriod(period), {
+                      locale: selectedLocale,
+                      relativeToTimestamp: 0,
+                      largestUnit: 'day',
+                    }),
+            }))}
+            value={isZooming ? '' : selectedPeriod}
+            onValueChange={(value) => setSelectedPeriod(value as TradingRewardsPeriod)}
+            onInteraction={onToggleInteract}
+          />
+        </$TitleContainer>
+      )}
+      <TimeSeriesChart
+        className={className}
+        selectedLocale={selectedLocale}
+        data={rewardsData}
+        yAxisOrientation="right"
+        margin={{
+          left: 32,
+          right: 48,
```

Comment:
not a real issue, but this chart is going to rerender a lot maybe since we are using now here and passing all these fresh objects

Reply:
hmm I think now should only update every `timeInterval` right now (which I guess I have set to a day ** I should probably update that to be more frequent to pick up new trading rewards, probably once every minute or so) so I don't think that should be too bad. But good point about fresh objects - lemme fix that